libraries/lookup-by-path/src/LookupByPath.ts

```diff
@@ -98,6 +98,27 @@ export interface IReadonlyLookupByPath<TItem> {
    */
   findChildPathFromSegments(childPathSegments: Iterable<string>): TItem | undefined;
 
+  /**
+   * Determines if an entry exists exactly at the specified path.
+   *
+   * @returns `true` if an entry exists at the specified path, `false` otherwise
+   */
+  has(query: string, delimiter?: string): boolean;
+
+  /**
+   * Retrieves the entry that exists exactly at the specified path, if any.
+   *
+   * @returns The entry that exists exactly at the specified path, or `undefined` if no entry exists.
+   */
+  get(query: string, delimiter?: string): TItem | undefined;
+
+  /**
+   * Determines if the trie is empty.
+   *
+   * @returns `true` if the trie is empty, `false` otherwise
+   */
+  isEmpty(): boolean;
```

Comment:
Could we replace this with a `count()` method instead? Seems like it would do similar things while being more useful. Would be cool to make this type be iterable as well.